src/Plivo/Util/signatureValidation.php

```diff
@@ -21,13 +21,13 @@ public static function validateSignature($uri, $nonce, $signature, $auth_token='
       $signature = utf8_encode($signature);
       $auth_token = utf8_encode($auth_token);
       $uri = utf8_encode($uri);
-
+      $port = '';
       $parsed_uri = parse_url($uri);
-
       $base_url = $parsed_uri['scheme'].'://'.$parsed_uri['host'];
-      if (isset($parsed_uri['path']))
-        $base_url = $parsed_uri['scheme'].'://'.$parsed_uri['host'].''.$parsed_uri['path'];
-
+      if (isset($parsed_uri['port'])) {
+        $base_url .= ':'.$parsed_uri['port'];
+      }
+      $base_url .= $parsed_uri['path'];
```

Comment:
Put a check for path as well

if (isset($parsed_uri['port']) && isset($parsed_uri['path']))
    $base_url = $parsed_uri['scheme'].'://'.$parsed_uri['host'].':'.$parsed_uri['port'].'/'.$parsed_uri['path'];
else if (isset($parsed_uri['path'])):
    $base_url = $parsed_uri['scheme'].'://'.$parsed_uri['host'].''.$parsed_uri['path'];